cloudformation-custom-resource/src/main/scala/feral/cloudformation/CloudFormationCustomResource.scala

```diff
@@ -0,0 +1,91 @@
+package feral.cloudformation
+
+import cats.effect._
+import cats.effect.kernel.Resource
+import cats.syntax.all._
+import feral.cloudformation.CloudFormationCustomResourceHandler.stackTraceLines
+import feral.cloudformation.CloudFormationRequestType._
+import feral.lambda.{Context, IOLambda}
+import io.circe._
+import io.circe.syntax._
+import org.http4s.Method.POST
+import org.http4s.client.Client
+import org.http4s.circe._
+import org.http4s.client.dsl.Http4sClientDsl
+import org.http4s.ember.client.EmberClientBuilder
+
+import java.io.{PrintWriter, StringWriter}
+
+trait CloudFormationCustomResource[F[_], Input, Output] {
+  def createResource(event: CloudFormationCustomResourceRequest[Input]): F[HandlerResponse[Output]]
+  def updateResource(event: CloudFormationCustomResourceRequest[Input]): F[HandlerResponse[Output]]
+  def deleteResource(event: CloudFormationCustomResourceRequest[Input]): F[HandlerResponse[Output]]
+}
+
+abstract class CloudFormationCustomResourceHandler[CfnSetup, Input : Decoder, Output: Encoder]
+  extends IOLambda[CloudFormationCustomResourceRequest[Input], Unit]
+    with Http4sClientDsl[IO] {
+  override protected type Setup = (Client[IO], CfnSetup)
+
+  def cfnSetup: Resource[IO, CfnSetup]
+  def handler(setup: CfnSetup): CloudFormationCustomResource[IO, Input, Output]
```

Comment:
I'm a little confused why we need these seperate. Is it possible to just ask for a `Resource[CloudFormationCustomResource[IO, Input, Output]`? Like I did in the http4s lambda, asking for a `Resource[HttpRoutes[IO]]` I think.

Reply:
How does the implementation of `Resource[CloudFormationCustomResource[IO, Input, Output]` get access to the materialized `CfnSetup`?